pkg/uhttp/wrapper.go

```diff
@@ -0,0 +1,107 @@
+package uhttp
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+type (
+	HttpClient interface {
+		HttpClient() *http.Client
+		Do(req *http.Request, options ...DoOption) (*http.Response, error)
+		NewRequest(ctx context.Context, method string, url *url.URL, options ...RequestOption) (*http.Request, error)
+	}
+	BaseHttpClient struct {
+		httpClient *http.Client
+	}
```

Comment:
@ggreer Not sure about the naming. Do you think it's fine?

Reply:
Seems fine.